src/librustc_query_system/lib.rs

```diff
@@ -16,6 +16,4 @@ extern crate rustc_data_structures;
 pub mod dep_graph;
 pub mod query;
 
-pub trait HashStableContext {
-    fn debug_dep_tasks(&self) -> bool;
-}
+pub trait HashStableContext {}
```

Comment:
This trait seems useless now =P

Reply:
It is used by the macro `HashStable_Generic` for `WorkProductId`. I can implement it by hand also.